src/verifier/SourcifyVerifier.ts

```diff
@@ -26,8 +26,17 @@ export class SourcifyVerifier extends AbstractVerifier implements Verifier {
     await this.checkBoundaries();
 
     const res = await this.sendVerifyRequest(artifact);
-    enforceOrThrow(res.data?.result?.length === 1, `Failed to connect to Sourcify API at url ${SOURCIFY_API_URL}`);
 
+    const verificationStatus = res.data?.result[0]?.status;
+    if (verificationStatus !== 'perfect' || verificationStatus !== 'partial') {
+      throw new Error(
+        `Sourcify couldn't verify the contract. Server response at ${SOURCIFY_API_URL}: \n ${JSON.stringify(
+          res.data.result[0],
```

Comment:
Is there any case in which `result[0]` is `undefined` or not JSON (causing the `JSON.stringify` to throw)?